pkg/autoops/domain/auto_ops_rule_test.go

```diff
@@ -26,21 +26,153 @@ import (
 )
 
 func TestNewAutoOpsRule(t *testing.T) {
-	t.Parallel()
-	aor := createAutoOpsRule(t)
-	assert.IsType(t, &AutoOpsRule{}, aor)
-	assert.Equal(t, "feature-id", aor.FeatureId)
-	assert.Equal(t, autoopsproto.OpsType_ENABLE_FEATURE, aor.OpsType)
-	assert.NotZero(t, aor.Clauses)
-	assert.Zero(t, aor.TriggeredAt)
-	assert.NotZero(t, aor.CreatedAt)
-	assert.NotZero(t, aor.UpdatedAt)
+	patterns := []struct {
+		desc             string
+		featureId        string
+		opsType          autoopsproto.OpsType
+		datetimeClauses  []*autoopsproto.DatetimeClause
+		eventRateClauses []*autoopsproto.OpsEventRateClause
+		expectedErr      error
+	}{
+		{
+			desc:             "OpsType: EnableFeature",
+			featureId:        "feature-id",
+			opsType:          autoopsproto.OpsType_ENABLE_FEATURE,
+			datetimeClauses:  []*autoopsproto.DatetimeClause{{Time: 0}},
+			eventRateClauses: []*autoopsproto.OpsEventRateClause{},
+		},
+		{
+			desc:            "OpsType: EnableFeature For EventRate",
```

Comment:
Please use `err` or `success` in the description to help us to identify what case it is.
Same for the other tests.

```suggestion
			desc:            "err: OpsType: EnableFeature For EventRate",
```